### PR TITLE
Sockets - Erica

### PR DESCRIPTION
**Pull request by norrise120**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The model is taking care of the _stuff_ related to the model.  The model handles talking to the database, keeping track of the data related to individual models.
Describe in your own words what the Controller is doing in Rails | The controller is acting as the intermediary between the user and the model.  
Describe in your own words what the View is doing in Rails | The view keeps track of what the user sees.  It's the interface with that the user uses to interact with the controller.
Describe an edge-case controller test you wrote | I wrote a test for cases in which the user provided an invalid id (like -1).
What is the purpose of using strong params? (i.e. the params method in the controller) | Strong params allow the controller to safely use data that comes from forms.  
How are Rails migrations related to Rails models? | Migrations allow changes to the Rails models.  For example, in my code, I used a migration to convert the datatype of completion_date from string to datetime.
Describe one area of Rails that are still unclear on | I have a feeling that I should have ignored some of the files when uploading to github.  The amount of files/lines changed seem must higher than it should be.  Should everything be tracked, or should some of the files be ignored?

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  end
+
+  def update
+    task_id = (params[:id])
+
+    @task = Task.find_by(id: task_id)
+
+    if @task
+      @task.name = params[:name]
+      @task.description = params[:description]
```
Comment (droberts-sea) [inline app/controllers/tasks_controller.rb:37]: This doesn't match how the name and description are stored in `params`. The result is that both `name` and `description` get set to `nil` here.

This is why manual testing is important in addition to automated testing - if the tests are wrong, your human eyeballs are the only way to catch the problem.

File: test/controllers/tasks_controller_test.rb [hunk 2]
```diff
@@ -90,13 +83,27 @@
...
-      # Your code here
+      #Arrange
+      task_data = {
+        name: "Edit test",
+        description: "build a test for the edit method",
+        completion_date: nil,
+      }
+
+      #Act
+      get edit_task_path(task.id), params: task_data
```
Comment (droberts-sea) [inline test/controllers/tasks_controller_test.rb:94]: You shouldn't need to send `params` along with the edit action - all you're doing here is loading the form, not submitting data yet.

File: test/controllers/tasks_controller_test.rb [hunk 3]
```diff
@@ -105,24 +112,118 @@
...
+      must_respond_with :redirect
+      must_redirect_to tasks_path
+
+      task = Task.find_by(id: id)
+      expect(task.completion_date).wont_be_nil
+    end
+
+    it "updates the task with nil completion time when unmarked complete" do
+      id = Task.first.id
+      task = Task.find_by(id: id)
```
Comment (droberts-sea) [inline test/controllers/tasks_controller_test.rb:214]: What happens if the task doesn't exist?

**Comment by droberts-sea** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | yes - 100 files seems like a lot, but this is about right. Rails is big!
Answered comprehension questions | yes
Successfully handles: Index, Show | yes
Index & Show tests pass | yes
Successfully handles: New, Create | yes, but your site doesn't contain a link to the `new` page, so I had to type in the URL manually. Not a great user experience.
New & Create tests pass | yes
Successfully handles: Edit, Update | no - see inline
Tests for Edit & Update test valid & invalid task ids |  yes
Successfully handles: Destroy, Task Complete | yes
Tests for Destroy & Task Complete include tests for valid and invalid task ids | yes
Routes follow RESTful conventions | yes. Any reason you're not using `resources`?
Uses named routes (like `_path`) | yes
**Overall** | This is a good start, and it feels like you've got the core learning goals around Rails. However there are some usability issues here that should have been caught before submission. Make sure you're running through your workflows manually in addition to writing solid tests.
